rest-api-spec/src/main/resources/rest-api-spec/test/index/100_partial_flat_object.yml

```diff
@@ -0,0 +1,539 @@
+---
+# The test setup includes:
+# - Create flat_object mapping for test_partial_flat_object index
+# - Index two example documents
+# - Refresh the index so it is ready for search tests
+
+setup:
+  - do:
+      indices.create:
+        index: test_partial_flat_object
+        body:
+          mappings:
+            properties:
+              issue:
+                properties:
+                  number:
+                    type : "integer"
+                  labels:
+                    type : "flat_object"
+  - do:
+      index:
+        index: test_partial_flat_object
+        id: 1
+        body: {
+          "issue": {
+            "number": 123,
+            "labels": {
+              "version": "2.2",
+              "backport": [
+                "2.0",
+                "1.9"
+              ],
+              "category": {
+                "type": "API",
+                "level": "bug"
+              },
+              "createdDate": "2023-01-01",
+              "comment" : [["Doe","Shipped"],["John","Approved"]],
+              "views": 288,
+              "priority": 5.00
+            }
+          }
+        }
+
+  - do:
+      index:
+        index: test_partial_flat_object
+        id: 2
+        body: {
+          "issue": {
+            "number": 456,
+            "labels": {
```

Comment:
Could you alter one of the documents as such that `labels` has a property (let say `author`) only one of those?  (and modify one of the exists queries to check that)

Reply:
added [#author](https://github.com/opensearch-project/OpenSearch/blob/3ec33621f9bbb38d5cbf56133397ec7c218d6dbd/rest-api-spec/src/main/resources/rest-api-spec/test/index/100_partial_flat_object.yml#L53) and test the [query](https://github.com/opensearch-project/OpenSearch/blob/3ec33621f9bbb38d5cbf56133397ec7c218d6dbd/rest-api-spec/src/main/resources/rest-api-spec/test/index/100_partial_flat_object.yml#L528)